packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -298,16 +317,67 @@ List<Plugin> findPlugins(FlutterProject project) {
     packages = PackageMap(packagesFile).map;
   } on FormatException catch (e) {
     printTrace('Invalid .packages file: $e');
-    return plugins;
+    return <Plugin>[];
   }
-  packages.forEach((String name, Uri uri) {
-    final Uri packageRoot = uri.resolve('..');
-    final Plugin plugin = _pluginFromPubspec(name, packageRoot);
-    if (plugin != null) {
-      plugins.add(plugin);
+
+  return packages.entries
+    .map((MapEntry<String, Uri> entry) {
+      final Uri packageRoot = entry.value.resolve('..');
+      return _packageFromPubspec(entry.key, packageRoot);
+    })
+    .where((_FoundPackage package) => package != null);
+}
+
+List<Plugin> findPlugins(FlutterProject project) {
+  final Iterable<_FoundPackage> allPackages = _findAllPackages(project);
+  final List<Plugin> allPlugins = allPackages.whereType<Plugin>().toList();
+
+  // if a pubspec file is available, only report plugins that aren't a dev
+  // dependency.
+  final String pubspecFile = fs.path.join(project.directory.path, 'pubspec.yaml');
+  if (!fs.isFileSync(pubspecFile)) {
+    return allPlugins;
+  }
+
+  final dynamic pubspec = loadYaml(fs.file(pubspecFile).readAsStringSync());
```

Comment:
Here, L343, and L354 you're looking up the dependencies from the pubspec and casting. Instead perform the lookup and cast once into a single local and reuse it.